src/components/molecules/NavSearchBar/NavSearchList.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { Key } from "react";
+import { FixedSizeList } from "react-window";
+
+import "./NavSearchList.scss";
+
+import {
+  NavSearchResultProps,
+  NavSearchResult,
+} from "components/molecules/NavSearchBar/NavSearchResult";
+
+// @debt refactor this into .scss or find more ergonomic component with same desired functionality as FixedSizeList
+// values are used here due to quirks of the FixedSizeList component
+// NOTE: keep in sync with the .scss files mentioned in comments
+const propsFor = (length: number) => ({
+  itemCount: length,
+  itemSize: 48, // 2 * $search-result-padding--vertical + $search-result-image--size; @see navSearchBarConstants.scss
+  height:
+    0 === length
+      ? 40 // $search-dropdown-height--empty; @see navSearchBarConstants.scss
+      : Math.min(
+          480, // $search-dropdown-height--max; @see NavSearchBar.scss
+          48 * length // 2 * $search-result-padding--vertical + $search-result-image--size; @see navSearchBarConstants.scss
+        ),
+  width: "18.125rem", // $search-dropdown-width; @see navSearchBarConstants.scss
+});
+
+export type NavSearchListItem = NavSearchResultProps & { key: Key };
+
+export interface NavSearchListProps {
+  items?: NavSearchListItem[];
+}
+
+export const NavSearchList: React.FC<NavSearchListProps> = ({ items = [] }) => {
+  const children = items.map(
+    ({ key, ...props }) => <NavSearchResult key={key} {...props} />,
+    [items]
+  );
+
+  return (
+    <FixedSizeList
+      className="NavSearchList NavSearchList__list"
+      {...propsFor(items.length)}
+    >
+      {({ index, style }) => (
+        <div
+          style={style}
+          key={items[index].key}
+          className="NavSearchList__item"
+        >
+          {children[index]}
+        </div>
+      )}
+    </FixedSizeList>
+  );
```

Comment:
```suggestion
  return (
    <FixedSizeList
      className="NavSearchList NavSearchList__list"
      {...propsFor(items.length)}
    >
      {({ index, style }) => {
        const {key, ...searchResultProps} = items[index];
        
        return (
          <div
            key={key}
            className="NavSearchList__item"
            style={style}
          >
            <NavSearchResult {...searchResultProps} />
          </div>
        )
      }}
    </FixedSizeList>
  );
```

Reply:
Is there any reason we need the wrapping div? Could we just directly render the `NavSearchResult` here?

Why applying both `NavSearchList` and `NavSearchList__list` to the container here? If they are the same thing, then it should all be in `NavSearchList` and we should just delete `NavSearchList__list`

Reply:
I add `NavSearchList` at the top level element of the component so that non-local style selectors can refer to children components e.g. `.NavSearchResult p` (or any element in `children`) while the `&_list` and `&_item` refer to known elements of the component itself.

The wrapping `div` is so that `react-window` will work its inline styles without interfering with the styles needed for the item components. Should `react-window` be replaced with something like https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API that `div` should be removed.

When I return to doing the required changes to this PR,  I will make an attempt of replacing with `IntersectionObserver` thus rendering the `div` addition and similar decisions moot.